core/application/bloodDonationWorkflow/BloodDonationService.ts

```diff
@@ -55,4 +59,52 @@ export class BloodDonationService {
       throw new BloodDonationOperationError(`Failed to update blood donation post. Error: ${error}`, GENERIC_CODES.ERROR)
     }
   }
+
+  async routeDonorRequest(
+    donorRoutingAttributes: DonorRoutingAttributes,
+    bloodDonationRepository: Repository<DonationDTO>,
+    stepFunctionModel: StepFunctionModel
+  ): Promise<string> {
+    try {
+      const { seekerId, requestPostId } = donorRoutingAttributes
+
+      const existingItem = await bloodDonationRepository.getItem(`${BLOOD_REQUEST_PK_PREFIX}#${seekerId}`, `${BLOOD_REQUEST_PK_PREFIX}#${requestPostId}`)
+      if (existingItem === null) {
+        return 'Item not found.'
+      }
+      if (existingItem.status === DonationStatus.COMPLETED || existingItem.status === DonationStatus.EXPIRED) {
+        return 'You can\'t update the donation request'
+      }
+
+      const retryCount = existingItem?.retryCount ?? 0
+      const updateData: Partial<DonationDTO> = {
+        ...existingItem,
+        id: requestPostId,
+        retryCount: retryCount + 1
+      }
+
+      if (retryCount >= MAX_RETRY_COUNT) {
+        updateData.status = DonationStatus.EXPIRED
+        await bloodDonationRepository.update(updateData)
+        return 'Donor search expired after max retries.'
```

Comment:
this sentence is strange, can we rephrase please

Reply:
working on it vai